apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-groups/resource-groups.component.ts

```diff
@@ -55,7 +55,22 @@ export class ResourceGroupsComponent implements OnInit {
         gws.sourceGroupId = g.sourceGroupId;
         return gws;
       });
-      this.groupsToDisable = new Set(this.assignedGroups.filter(group => !!group.sourceGroupId).map(group => group.id));
+
+      // identify groups with multiple assignments
+      const groupsIds = this.assignedGroups.map(group => group.id);
+      const groupsWithMultiAssignments = this.assignedGroups.filter((group, i) =>
+        groupsIds.indexOf(group.id) !== i && (group.sourceGroupId === null || this.assignedGroups[groupsIds.indexOf(group.id)].sourceGroupId === null)).map(group => group.id);
+      const uniqueGroups = this.assignedGroups.filter((group, i) => groupsIds.indexOf(group.id) === i);
```

Comment:
Please, put the main `.filter` action in a block and format it a little so it's not just a super long line. Also don't calculate `groupsIds.indexOf(group.id)` twice.